### PR TITLE
Fix issue where the Email API crashes if an attempt is made to proces…

### PR DESCRIPTION
**Pull request by andrew-cormick-dockery**
…s tokens against a contact whose email is set to On Hold

This is a fix that @ineffyble suggested in relation to #HELP-1825, where the Email API crashed trying to extract information to populate tokens of contacts whose email address is On Hold.

We now ensure that token processing does not exclude contacts whose address is On Hold.  This is not necessarily related to the target email address, which is constrained from sending to On Hold email addresses via other mechanisms.  The specific use case targeted by this fix is to extract contact information about a contact whose email address is on hold and then sending that information to a completely different email address.  An example might be to send an alert email to an administrator that a problem has occurred with a contact.

### CODE REVIEW

**Review by seamuslee001 — APPROVED** [review]

File: api/v3/Email/Send.php
Comment (ineffyble) [inline api/v3/Email/Send.php]: I am of the view that we should disable the deceased filter too, since these crashes can cause major problems, and [line 151](https://github.com/australiangreens/org.civicoop.emailapi/pull/6/files#diff-d1c70eeb386beae24e3cc3a629071ab92ec88a15cefda8580e81f6ac1182df6bR151) already ensures that we will never send an email if the recipient is deceased.

```suggestion
    $contactHookArray = CRM_Utils_Token::getTokenDetails([$contact['contact_id']], NULL, FALSE, FALSE)[0];
```

**Comment by ineffyble** [comment]
Note that this is not just for On Hold. The same issue occurs if a contact has no email address.

**Comment by seamuslee001** [comment]
I would also suggest that we look at implementing #4 as a longer term approach as it uses token processor and is from upstream

**Comment by ineffyble** [comment]
> I would also suggest that we look at implementing #4 as a longer term approach as it uses token processor and is from upstream

Yes. That will also unblock us from refactoring melons tokens to stop using the legacy hooks.

**Review by johntwyman — APPROVED** [review]

**Pull request merged**
